skada/_reweight.py

```diff
@@ -671,6 +673,145 @@ def KLIEP(
     )
 
 
+class NearestNeighborDensityAdapter(BaseAdapter):
+    """Adapter based on re-weighting samples using a 1NN,
+
+    See: [Loog, 2012] Loog, M. (2012).
+    Nearest neighbor-based importance weighting.
+    In 2012 IEEE International Workshop on Machine
+    Learning for Signal Processing, pages 1–6. IEEE.
+
+    Parameters
+    ----------
+    base_estimator : estimator object, optional
+        The estimator to use to estimate the densities of source and target
+        observations. If None, a KNeighborsClassifier(n_neighbors=1) estimator
+        is used.
+
+    Attributes
+    ----------
+    weight_estimator_source_ : object
+        The estimator object fitted on the source data.
+    weight_estimator_target_ : object
+        The estimator object fitted on the target data.
+    """
+
+    def __init__(self, base_estimator=None, laplace_smoothing=False):
+        super().__init__()
+        self.laplace_smoothing = laplace_smoothing
+        self.base_estimator = base_estimator or KNeighborsClassifier(
+            n_neighbors=1)
+
+    def fit(self, X, y=None, sample_domain=None):
+        """Fit adaptation parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, sample_domain = check_X_domain(X, sample_domain)
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+        self.X_source_fit = X_source
+        indices_source = np.arange(X_source.shape[0])
+
+        self.estimator_ = clone(self.base_estimator)
+        self.estimator_.fit(X_source, indices_source)
+
+        return self
+
+    def get_weights(self, Xs, Xt):
+        indices_source = np.arange(Xs.shape[0])
+        estimator = clone(self.base_estimator)
+        estimator.fit(Xs, indices_source)
+        predictions = estimator.predict(Xt)
+        a = (1 if self.laplace_smoothing else 0)
+        weights = np.array(
+            [a + np.count_nonzero(predictions == i) for i in indices_source])
```

Comment:
Maybe you can fasten this part of the implementation by using `np.unique`, with something like:

```pyhton
unique, counts = np.unique(predictions, return_counts=True)
weights = np.ones(Xs.shape[0]) * float(self.laplace_smoothing)
weights[unique] += counts
```

Reply:
Hi @BuenoRuben,
I see that you mark this review as resolved, but it was not linked with my previous (not relevant) reviews for this part of the code. I still believe that using `np.unique` here is faster...